packages/shared/src/hooks/useBookmarkProvider.ts

```diff
@@ -0,0 +1,43 @@
+import { useEffect, useState } from 'react';
+import { feature } from '../lib/featureManagement';
+import { useActiveFeedNameContext } from '../contexts';
+import { SharedFeedPage } from '../components/utilities';
+import { useConditionalFeature } from './useConditionalFeature';
+
+interface UseBookmarkProviderReturn {
+  highlightBookmarkedPost: boolean;
+}
+
+interface UseBookmarkProviderProps {
+  bookmarked?: boolean;
+}
+
+const useBookmarkProvider = ({
+  bookmarked = false,
+}: UseBookmarkProviderProps): UseBookmarkProviderReturn => {
+  const [isBookmarked] = useState<boolean>(bookmarked);
+  const [justBookmarked, setJustBookmarked] = useState<boolean>(false);
+  const { feedName } = useActiveFeedNameContext();
+
+  useEffect(() => {
+    if (bookmarked !== isBookmarked) {
+      setJustBookmarked(bookmarked);
+    }
+  }, [bookmarked, justBookmarked, setJustBookmarked, isBookmarked]);
```

Comment:
We tend to avoid use effects whenever possible, however, this has made the process much simpler by not injecting the state update on bookmark click. Would love to hear more thoughts here.

Reply:
I don't think we actually need useEffect and the state justBookmarked at all, is useless, we can just compare the values directly. 
was a bit tired yesterday not sure what was I thinking 😓